fdbserver/MockLogSystem.h

```diff
@@ -28,6 +28,7 @@ struct MockLogSystem : ILogSystem, ReferenceCounted<MockLogSystem> {
 	Reference<ILogSystem::IPeekCursor> cursor;
 
 	MockLogSystem();
+	MockLogSystem(MockLogSystem& that);
```

Comment:
Is this needed? If so, it is convention for copy constructors to accept a const reference. You should also implement the copy assignment operator and either implement or disable the move constructor and move assignment operator.

Reply:
Copy constructor is needed here https://github.com/apple/foundationdb/pull/4734/files#diff-c4dbfae809f757f2061099ee5021ce2cbbf0f4dbc2f0f18480a4856d63d033ceR4611. We need to defined the copy constructor explicitly because the implicit constructor of `MockLogSystem` requires the implicit constructor of `ReferenceCounted`, which is deleted.

I'm adding
```
    MockLogSystem(const MockLogSystem&& that) = delete;
    MockLogSystem& operator=(const MockLogSystem&) = delete;
    MockLogSystem& operator=(const MockLogSystem&&) = delete;
```
to disable the move constructor and both assignment operator.

Reply:
Please implement the copy assignment operator. A type with a copy constructor but not a copy assignment operator violates developers' expectations. It also prevents initialization of the form:

```
MockLogSystem foo = {bar, baz};
```

even though that code will never actually call the copy assignment operator.